<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="ziweil8" />

<meta name="date" content="2018-09-03" />

<title>Reproduce output of a paper</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">live-demo</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Reproduce output of a paper</h1>
<h4 class="author"><em>ziweil8</em></h4>
<h4 class="date"><em>2018-09-03</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-09-08</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20180717)</code> </summary></p>
<p>The command <code>set.seed(20180717)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/ziweil8/live-demo/tree/116d5a2396c44912f3259ce012089d08b6d0c07e" target="_blank">116d5a2</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/ziweil8/live-demo/2a98b5381d96b4a348e44f7bc907efe9b95a9c62/docs/first-analysis.html" target="_blank">2a98b53</a>
</td>
<td style="text-align:left;">
ziweil8
</td>
<td style="text-align:left;">
2018-09-08
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/ziweil8/live-demo/blob/ea07f25b8ef12605907897ba6cb2f6abb750d89a/analysis/first-analysis.Rmd" target="_blank">ea07f25</a>
</td>
<td style="text-align:left;">
ziweil8
</td>
<td style="text-align:left;">
2018-09-08
</td>
<td style="text-align:left;">
update
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/ziweil8/live-demo/9f1a863e1b9242771832709b14a5e652ef0bb8b6/docs/first-analysis.html" target="_blank">9f1a863</a>
</td>
<td style="text-align:left;">
ziweil8
</td>
<td style="text-align:left;">
2018-09-08
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/ziweil8/live-demo/blob/fb9b1a75140b61f0e8f9a643531c1a86815f58e2/analysis/first-analysis.Rmd" target="_blank">fb9b1a7</a>
</td>
<td style="text-align:left;">
ziweil8
</td>
<td style="text-align:left;">
2018-09-08
</td>
<td style="text-align:left;">
update
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<hr />
<div id="preprocess-data" class="section level2">
<h2>preprocess data</h2>
<div id="download-data" class="section level3">
<h3>download data</h3>
<p>Website of sra data<a href="https://www.ncbi.nlm.nih.gov/sra/?term=SRP049743">link</a></p>
<pre class="r"><code>module load SRA-Toolkit/2.8.2-1-centos_linux64
prefetch -v SRRXXX </code></pre>
<p>the sra data path:/vlsci/SG0010/ziwei/ncbi/public/sra</p>
</div>
<div id="convert-sra-to-fastq" class="section level3">
<h3>Convert sra to fastq</h3>
<pre class="r"><code>module load SRA-Toolkit/2.8.2-1-centos_linux64 
fastq-dump --split-files SRRXXX.sra </code></pre>
</div>
<div id="move-fastq-to-another-path" class="section level3">
<h3>Move fastq to another path</h3>
<pre class="r"><code>mv *.fastq/vlsci/SG0010/shared/shared_data/external_public/Russ_2017/raw/fastq</code></pre>
</div>
<div id="download-fasta" class="section level3">
<h3>Download fasta</h3>
<pre class="r"><code>wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.26_GRCm38.p6/GCF_000001635.26_GRCm38.p6_genomic.fna.gz 

gunzip GCF_000001635.26_GRCm38.p6_genomic.fna.gz</code></pre>
<p>the fasta data path:/vlsci/SG0010/shared/shared_data/external_public/Russ_2017/raw/fasta</p>
</div>
<div id="build-index" class="section level3">
<h3>build index</h3>
<pre class="r"><code>bowtie2-build refseq/*.file mm10_index/mm10_index </code></pre>
<hr />
</div>
</div>
<div id="mapping" class="section level2">
<h2>Mapping</h2>
<div id="create-scirpt" class="section level3">
<h3>create scirpt</h3>
<p>create script with the help of Job script generator for SLURM[<a href="https://www.melbournebioinformatics.org.au/jobscript-generator/" class="uri">https://www.melbournebioinformatics.org.au/jobscript-generator/</a>] details about some parameters Job Type:SMP CPU Cores:12 Memory in Gigabytes:100 hours:23 Modules: Bowtie2/2.3.2-vlsci_intel-2015.08.25</p>
<pre class="r"><code>module load Bowtie2/2.3.2-vlsci_intel-2015.08.25
bowtie2 [options]* -x &lt;bt2-idx&gt; {-1 &lt;m1&gt; -2 &lt;m2&gt; | -U &lt;r&gt; | --interleaved &lt;i&gt;} -S [&lt;sam&gt;]</code></pre>
<p>for example</p>
<pre class="r"><code>bowtie2 -x raw/fasta/index/mm10 -U raw/fastq/SRR1772989_1.fastq  -S output/SRR1772989_Chip-seq.sam   </code></pre>
</div>
<div id="new-a-sbatch-file" class="section level3">
<h3>new a sbatch file</h3>
<pre class="r"><code>vi filename.sbatch</code></pre>
<p>cope all content of script to sbatch file and then press Esc (for exiting insertion statement)</p>
<p>and then save and exit sbatch file</p>
<pre class="r"><code>:wq ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.26_GRCm38.p6/GCF_000001635.26_GRCm38.p6_genomic.fna.gz</code></pre>
</div>
<div id="submit-sbatch-file" class="section level3">
<h3>submit sbatch file</h3>
<pre class="r"><code>sbatch filenam.sbatch</code></pre>
</div>
<div id="view-waiting-statement" class="section level3">
<h3>view waiting statement</h3>
<pre class="r"><code>showq -u ziwei
or 
squeue -u ziwei</code></pre>
</div>
<div id="path-of-mapping-output" class="section level3">
<h3>path of mapping output</h3>
<pre class="r"><code>/vlsci/SG0010/shared/shared_data/external_public/Russ_2017/output/sam</code></pre>
<hr />
</div>
</div>
<div id="sam-to-bam" class="section level2">
<h2>sam to bam</h2>
<pre class="r"><code>module load SAMtools/1.5-vlsci_intel-2015.08.25-HTSlib-1.5
samtools view -S -b sample.sam &gt; sample.bam</code></pre>
<div id="filter-reads-with-low-alignment-quality-10" class="section level3">
<h3>Filter reads with low alignment quality 10</h3>
<pre class="r"><code>samtools view -b -q 10 output/bam/SRR1772989.bam &gt; output/bam/filteredbam/SRR1772989.qfiltered10.bam</code></pre>
<hr />
</div>
</div>
<div id="bam-to-hdf5" class="section level2">
<h2>bam to hdf5</h2>
<div id="index-filtered-bam-files" class="section level3">
<h3>Index filtered bam files</h3>
<pre class="r"><code>samtools sort -o sorted.bam initial.bam
samtools index sorted.bam</code></pre>
</div>
<div id="convert" class="section level3">
<h3>convert</h3>
<pre class="r"><code>python load_bam_5prime_ends.py --assembly hg19 /SRR1772989sorted.fwd /SRR1772989sorted.rev /sortedbam/SRR1772989sorted.bam</code></pre>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.4 (2018-03-15)
Platform: i386-w64-mingw32/i386 (32-bit)
Running under: Windows 7 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
[3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] workflowr_1.1.1   Rcpp_0.12.16      digest_0.6.15    
 [4] rprojroot_1.3-2   R.methodsS3_1.7.1 backports_1.1.2  
 [7] git2r_0.22.1      magrittr_1.5      evaluate_0.10.1  
[10] stringi_1.1.7     whisker_0.3-2     R.oo_1.22.0      
[13] R.utils_2.6.0     rmarkdown_1.10    tools_3.4.4      
[16] stringr_1.3.1     yaml_2.1.19       compiler_3.4.4   
[19] htmltools_0.3.6   knitr_1.20       </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
